homeassistant/components/dsmr/sensor.py

```diff
@@ -232,8 +238,14 @@ def unit_of_measurement(self):
         return self.get_dsmr_object_attr("unit")
 
     @staticmethod
-    def translate_tariff(value):
-        """Convert 2/1 to normal/low."""
+    def translate_tariff(value, dsmr_version):
+        """Convert 2/1 to normal/low depening on DSMR version."""
+        # In Belgium 0001 is normal, 0002 is low
+        if dsmr_version in ("5B"):
+            if value == "0001":
+                return "normal"
+            if value == "0002":
+                return "low"
```

Comment:
You could swap the values instead of writing more return statements, e.g.
```python
if value == "0001":
    value = "0002"
if value == "0002":
    value = "0001"
```

Reply:
Is there a reason you prefer this?
Cause your example will always give 0001 as return :)

Reply:
Yeah there is: you do not write duplicate return statements with the exact same return value.

> Cause your example will always give 0001 as return :)

Not sure what you mean by that. I am swapping the values so that the correct return value is chosen below.

Reply:
First if, you set value to 0002, and then the second if matches, and it sets it again to 0001 :)
Fixed via elif

Reply:
Yeah you are right 😁